src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Results/CompositionResult.cs

```diff
@@ -0,0 +1,54 @@
+using HotChocolate.Fusion.Errors;
+
+namespace HotChocolate.Fusion.Results;
+
+public readonly record struct CompositionResult
+{
+    public bool IsFailure { get; }
```

Comment:
Is there a concept in the spec of "Success with warnings" that should be supported?

Reply:
The spec has codes and severities, which we apply to entries in the `CompositionLog`.

We use the Result type instead of exceptions for our implementation, but this is not something that is specified.

If a validation rule, for example, logs a warning, it will still return a Success result. It will only return a Failure result if at least 1 error is logged in the composition log.